File: src/prod/src/ServiceModel/management/FileStoreService/uploadsessioninfo.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace Management::FileStoreService;

UploadSessionInfo::UploadSessionInfo()
: storeRelativeLocation_()
, sessionId_()
, fileSize_()
, modifiedDate_()
, expectedRanges_()
{
}

UploadSessionInfo::UploadSessionInfo(
    std::wstring storeRelativeLocation,
    Common::Guid sessionId,
    uint64 fileSize,
    Common::DateTime modifiedDate,
    std::vector<ChunkRangePair> && expectedRanges)
    : storeRelativeLocation_(storeRelativeLocation)
    , sessionId_(sessionId)
    , fileSize_(fileSize)
    , modifiedDate_(modifiedDate)
    , expectedRanges_(std::move(expectedRanges))
{
}

UploadSessionInfo::UploadSessionInfo(
    UploadSessionInfo && other)
    : storeRelativeLocation_(move(other.storeRelativeLocation_))
    , sessionId_(move(other.sessionId_))
    , fileSize_(other.fileSize_)
    , modifiedDate_(move(other.modifiedDate_))
    , expectedRanges_(move(other.expectedRanges_))
{
}

UploadSessionInfo const & UploadSessionInfo::operator = (UploadSessionInfo && other)
{
    if (this != &other)
    {
        this->storeRelativeLocation_ = move(other.storeRelativeLocation_);
        this->sessionId_ = move(other.sessionId_);
        this->fileSize_ = other.fileSize_;
        this->modifiedDate_ = move(other.modifiedDate_);
        this->expectedRanges_ = move(other.expectedRanges_);
    }

    return *this;
}

UploadSessionInfo::UploadSessionInfo(
    UploadSessionInfo const & other)
    : storeRelativeLocation_(other.storeRelativeLocation_)
    , sessionId_(other.sessionId_)
    , fileSize_(other.fileSize_)
    , modifiedDate_(other.modifiedDate_)
    , expectedRanges_(other.expectedRanges_)
{
}

UploadSessionInfo const & UploadSessionInfo::operator = (UploadSessionInfo const & other)
{
    if (this != &other)
    {
        this->storeRelativeLocation_ = other.storeRelativeLocation_;
        this->sessionId_ = other.sessionId_;
        this->fileSize_ = other.fileSize_;
        this->modifiedDate_ = other.modifiedDate_;
        this->expectedRanges_ = other.expectedRanges_;
    }

    return *this;
}

void UploadSessionInfo::WriteTo(TextWriter & w, FormatOptions const &) const
{
	w.Write("UploadSessionInfo { ");
    w.Write("StoreRelativeLocation = {0}", this->storeRelativeLocation_);
    w.Write("SessionId = {0}", this->sessionId_);
    w.Write("FileSize = {0}", this->fileSize_);
    w.Write("ModifiedDate = {0}", this->modifiedDate_.ToString());
    w.Write("ExpectedRanges { ");
    for (auto it = this->expectedRanges_.begin(); it != this->expectedRanges_.end(); ++it)
    {
        w.Write("StartPosition = {0}", it->StartPosition);
        w.Write("EndPosition = {0}", it->EndPosition);
    }

    w.Write("}");
    w.Write("}");
}

File: src/prod/src/managed/Api/src/System/Fabric/query/UploadSessionInfo.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Query
{
    using System.Fabric;
    using System.Fabric.Common.Serialization;
    using System.Fabric.Health;
    using System.Fabric.Interop;
    using System.Numerics;

    /// <summary>
    /// Contains information about the image store upload session
    /// </summary>
    public sealed class UploadSessionInfo
    {
        /// <summary>
        /// Initializes a new instance of the <cref name="UploadSessionInfo"/> structure
        /// </summary>
        /// <param name="storeRelativePath">
        /// Get the relative path of image store file which is from the native image store root.
        /// This parameter cannot be empty, or consist only of whitespace
        /// </param>
        /// <param name="sessionId">
        ///  Gets the upload session ID, this session ID cannot be used for two different relative path concurrently. 
        ///  Only if the old session was committed or removed by system, the old session ID can be reused in the new session.
        ///  This parameter cannot be empty
        /// </param>
        /// <param name="modifiedDate">
        /// Gets the date and time when the file was last modified
        /// This parameter cannot be empty
        /// </param>
        /// <param name="fileSize">
        /// Gets the size in byte of the native image store file
        /// This parameter cannot be empty
        /// </param>
        /// <param name="expectedRange">
        /// Get the range of the uploading file that have not been received
        /// </param>
        internal UploadSessionInfo(
            string storeRelativePath,
            Guid sessionId,
            DateTime modifiedDate,
            Int64 fileSize,
            UploadChunkRange[] expectedRange)
        {
            this.StoreRelativePath = storeRelativePath;
            this.SessionId = sessionId;
            this.ModifiedDate = modifiedDate;
            this.FileSize = fileSize;
            this.ExpectedRanges = expectedRange;
        }

        /// <summary>
        /// Initializes a new instance of the <cref name="UploadSessionInfo"/> structure
        /// </summary>
        internal UploadSessionInfo() { }

        /// <summary>
        /// <para>Get the relative path of the file from the native image store root.</para>
        /// </summary>
        /// <value>
        /// <para>The relative path of the file from the native image store root.</para>
        /// </value>
        public string StoreRelativePath { get; private set; }

        /// <summary>
        /// <para>Gets the upload session ID.</para>
        /// </summary>
        /// <value>
        /// <para>The upload session ID.</para>
        /// </value>
        public Guid SessionId { get; private set; }

        /// <summary>
        /// <para>Gets the date and time when the file was last modified.</para>
        /// </summary>
        /// <value>
        /// <para>The date and time when the file was last modified.</para>
        /// </value>
        public DateTime ModifiedDate { get; private set; }

        /// <summary>
        /// <para>Gets the size in bytes of the image store file.</para>
        /// </summary>
        /// <value>
        /// <para>The size in bytes of the image store file.</para>
        /// </value>
        public Int64 FileSize { get; private set; }

        /// <summary>
        ///<para>Get the range of the uploading file that have not been received.</para>
        /// </summary>
        /// <value>
        /// <para>The range of uploading file that have not been received.</para>
        /// </value>
        public UploadChunkRange[] ExpectedRanges { get; private set; }
    }
}